x-pack/plugins/infra/server/routes/snapshot/lib/apply_metadata_to_last_path.ts

```diff
@@ -47,12 +47,12 @@ export const applyMetadataToLastPath = (
         source.configuration.fields
       );
       // Set the label as the name and fallback to the id OR path.value
-      lastPath.label = get(firstMetaDoc, inventoryFields.name, lastPath.value);
+      lastPath.label = (firstMetaDoc[inventoryFields.name] as string | undefined) ?? lastPath.value;
```

Comment:
I would just leave this as it was which will fix your type error after the type change suggestion is updated